compact.go

```diff
@@ -659,50 +665,68 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 			continue
 		}
 
-		for i, chk := range chks {
-			if chk.MinTime < meta.MinTime || chk.MaxTime > meta.MaxTime {
-				return errors.Errorf("found chunk with minTime: %d maxTime: %d outside of compacted minTime: %d maxTime: %d",
-					chk.MinTime, chk.MaxTime, meta.MinTime, meta.MaxTime)
+		for i := 0; i < len(chks); i++ {
```

Comment:
Yup, so as you can see we remove elements from the `chks` while iterating.

It is safe to do, however `i` (current iterator) needs to be decremented (as the current element was removed). This is not possible in `for i := range array` since `i` is always incremented. Move to this allows for flexibility.

Reply:
yep, noted. I remember hitting this scenario few times.